packages/xo-server/src/api/vif.js

```diff
@@ -118,7 +118,8 @@ export async function set({
     const newVif = await xapi.createVif(vm.$id, network.$id, {
       mac,
       currently_attached: attached,
-      ipv4_allowed: newIpAddresses,
+      ipv4_allowed:
+        network.id !== vif.$network ? allowedIpv4Addresses : newIpAddresses,
```

Comment:
When the network is being set, the IP addresses aren't properly allocated in the Self Service (line 133).